src/appleseed.studio/mainwindow/consolewidget.cpp

```diff
@@ -55,65 +58,76 @@ ConsoleWidget::ConsoleWidget(QWidget* parent)
-
-    this->insertWidget(0, output);
-    this->insertWidget(1, input);
-    this->setOrientation(Qt::Vertical);
+    input = new PythonInput(this);
 
-    m_action_execute_selection = new QAction("Execute selection as python script", this);
+    m_action_execute_selection = new QAction(this);
     m_action_execute_selection->setShortcut(Qt::CTRL + Qt::Key_Return);
     m_action_execute_selection->setShortcutContext(Qt::WidgetWithChildrenShortcut);
+    m_action_execute_selection->setText("ES");
+    m_action_execute_selection->setToolTip("Execute selection as python script");
 
     connect(m_action_execute_selection, SIGNAL(triggered()), this, SLOT(slot_execute_selection()));
     addAction(m_action_execute_selection);
 
-    m_action_execute_all = new QAction("Execute all text in input as python script", this);
+    m_action_execute_all = new QAction(this);
     m_action_execute_all->setShortcut(Qt::CTRL + Qt::SHIFT + Qt::Key_Return);
     m_action_execute_all->setShortcutContext(Qt::WidgetWithChildrenShortcut);
+    m_action_execute_all->setText("EA");
+    m_action_execute_all->setToolTip("Execute all text in input as python script");
 
     connect(m_action_execute_all, SIGNAL(triggered()), this, SLOT(slot_execute_all()));
     addAction(m_action_execute_all);
 
+    m_action_clear_selection = new QAction(this);
+    m_action_clear_selection->setShortcut(Qt::CTRL + Qt::Key_D);
+    m_action_clear_selection->setShortcutContext(Qt::WidgetWithChildrenShortcut);
+    m_action_clear_selection->setText("CO");
+    m_action_clear_selection->setToolTip("Clear console output");
+
+    connect(m_action_clear_selection, SIGNAL(triggered()), this, SLOT(slot_clear_output()));
+    addAction(m_action_clear_selection);
+
     m_action_focus_on_input = new QAction("Focus on console input", this);
     m_action_focus_on_input->setShortcut(Qt::CTRL + Qt::Key_L);
 
     connect(m_action_focus_on_input, SIGNAL(triggered()), input, SLOT(setFocus()));
     addAction(m_action_focus_on_input);
 
+    QToolBar* toolBar = new QToolBar();
+    toolBar->addAction(m_action_execute_selection);
+    toolBar->addAction(m_action_execute_all);
+    toolBar->addAction(m_action_clear_selection);
+
+    this->insertWidget(0, toolBar);
+    this->insertWidget(1, output);
+    this->insertWidget(2, input);
+    this->setOrientation(Qt::Vertical);
+
     PythonInterpreter::instance().redirect_output(OutputRedirector(output));
 }
 
 void ConsoleWidget::slot_execute_selection()
 {
-    // QTextCursor returned by textCursor function use QChar(8233) istead of newline
+    // QTextCursor returned by textCursor function use QChar(8233) instead of newline
     // It breaks Python indentation rules so it has to be replaced
     QString selected = input->textCursor().selectedText().replace(QChar(8233), "\n");
     execute(selected);
 }
 
 void ConsoleWidget::slot_execute_all()
 {
-    // QTextCursor returned by textCursor function use QChar(8233) istead of newline
```

Comment:
It was there until I realised that this weird replacement is only in textCursor().selectedText() but not here